strapi/strapi
# enhancement: make policy error public

### What does it do?

PolicyError messages will be public to the API [as originally intended](https://github.com/strapi/strapi/pull/12080)

### Why is it needed?

PolicyError was created to allow policies to throw errors with messages visible to the user. Unfortunately, it was broken for all (or nearly all) of v4, so it was considered a security risk to suddenly make them visible at that point.

In v5, we will add the originally itended behavior to allow policies to throw a custom error message.

### How to test it?

Create a policy that throws a custom error message and you should see it
Any other type of ForbiddenError should still be caught and made generic (ie, the message should not be included in the api response)

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request
